tests/e2e/common/retry_test_mixins.py

```diff
@@ -15,8 +32,10 @@ def test_client_should_not_retry_429_if_RateLimitRetry_is_0(self):
                     rows = cursor.fetchall()
                     self.assertEqual(len(rows), 1)
                     self.assertEqual(rows[0][0], 1)
-        expected = "Maximum rate of 1 requests per SECOND has been exceeded. " \
-                   "Please reduce the rate of requests and try again after 1 seconds."
+        expected = (
```

Comment:
is there anyway we can test with behavior rather than log verification?  Log verification tends to be a very brittle testing approach.

Reply:
The test you commented on will be removed in v3. We never run it today. It only appears changed here because `black` caught it in the format step.

That said, in the new tests I do perform a log verification because what we're testing is that (1) the connector stops retrying when it receives a 404 (2) The connector logs that this happened. I agree that asserting logs is brittle when the log message is coming from an upstream dependency. But since the log message we're asserting in my test down below is generated _by this connector_ it's not brittle in that way.